scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,438 @@
+from lxml.etree import QName
+
+minigraph_ns = "Microsoft.Search.Autopilot.Evolution"
+minigraph_ns1 = "http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution"
+
+DEFAULT_DEV_PATH = '/usr/share/sonic/device/'
+
+### port_config.ini header
+PORTCONFIG_HEADER = ["# name", "lanes", "speed", "alias", "index"]
+
+
+class SkuCreate(object):
+	"""
+	Process aclstat
+	"""
+
+	def __init__(self):
+
+		self.portconfig_dict = {}
+		self.platform_specific_dict = {"x86_64-mlnx_msn2700-r0":self.msn2700_specific}
+		self.default_lanes_per_port = []
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
+		
+
+		try:
+			platform = subprocess.check_output("sonic-cfggen -H -v DEVICE_METADATA.localhost.platform",shell=True) #self.metadata['platform']
+			self.platform = platform.rstrip()
+		except KeyError:
+			print ("Couldn't find platform info in CONFIG_DB DEVICE_METADATA", file=sys.stderr)
+			exit(1)
+		self.sku_name = None
+		self.base_sku_name = None
+		self.base_sku_dir = None
+		self.base_file_path = None		
+		self.new_sku_dir = None
+		self.verbose = None
+
+	def read_metadata(self):
+		"""
+		Redis database platform info
+		"""
+		try:
+			self.metadata = self.db.get_all(self.db.CONFIG_DB, "DEVICE_METADATA|localhost" )
+		except:
+			print ("Error While trying to retrieve METADATA from CONFIG_DB", file=sys.stderr)
+			exit(1)
+			
+	def sku_def_parser(self,sku_def) :
+		try:
+			f = open(str(sku_def),"r")
+		except IOError:
+			print ("Couldn't open file: " + str(sku_def), file=sys.stderr)
+			exit(1)
+		element = ET.parse(f)
+		
+		root = element.getroot()
+		if (self.verbose):
+			print ( "tag=%s, attrib=%s" % (root.tag, root.attrib))
+		self.sku_name = root.attrib["HwSku"]
+		self.new_sku_dir = DEFAULT_DEV_PATH+self.platform+"/"+self.sku_name+ '/'
+		idx = 1
+		for child in root:
+			if child.tag == "Ethernet":
+				for interface in child:
+					for eth_iter in interface.iter():
+						if eth_iter is not None:
+							self.portconfig_dict[idx] = ["Ethernet"+eth_iter.get("Index"),[1,2,3,4], eth_iter.get("Speed"), eth_iter.get("InterfaceName"),  eth_iter.get("Index")]
+							if (self.verbose):
+								print ("sku_def_parser:portconfig_dict[",idx,"] -> ",self.portconfig_dict[idx])
+							idx += 1
+
+		f.close()
+
+	def parse_deviceinfo(self,meta,hwsku):
+		idx = 1
+		match = None
+		for device_info in meta.findall(str(QName(minigraph_ns, "DeviceInfo"))):
+			dev_sku = device_info.find(str(QName(minigraph_ns, "HwSku"))).text
+			if dev_sku == hwsku :
+				match = True
+				interfaces = device_info.find(str(QName(minigraph_ns, "EthernetInterfaces"))).findall(str(QName(minigraph_ns1, "EthernetInterface")))
+				for interface in interfaces:
+					alias = interface.find(str(QName(minigraph_ns, "InterfaceName"))).text
+					speed = interface.find(str(QName(minigraph_ns, "Speed"))).text
+					index  = interface.find(str(QName(minigraph_ns, "Index"))).text
```

Comment:
>"Index" [](start = 53, length = 7)

The minigraph parser is reading "Index" column, which is not needed by your algorithm. Please remove.

Reply:
Removed the usage of parsing the "Index" column form xml and minigraph files